src/renderer/components/+catalog/catalog.tsx

```diff
@@ -92,24 +92,34 @@ export class Catalog extends React.Component {
   }
 
   @action
-  onTabChange = (tabId: string) => {
-    this.activeTab = tabId;
+  onTabChange = (tabId: string | null) => {
```

Comment:
Why `tabId` is allowed to be `null`?
It looks like if it is null, `this.activeTab` is going to get reassigned to `undefined` - this looks like potential UB.

Reply:
UB doesn't exist in JS... I did this because the ID of a tab normally has to be the filtering type, and I chose `null` since that is not a string. I guess I could have chosen `true` or some other marker value.

Reply:
I guess I used the term incorrectly, sorry. What I meant is that one of potential results of this function now is assigning `undefined` to a value, the type definition of which doesn't allow `undefined`. You'll see what I mean if you add `"strictNullChecks": true,` to `tsconfig.json` (which should already have been there in the first place IMHO):
![Screen Shot 2021-04-30 at 1 32 36 PM](https://user-images.githubusercontent.com/22235527/116683876-dce4f080-a9b8-11eb-88c0-c3746bf2b17c.png)

If this is intended, it should be reflected in the type:
```ts
// line 31
@observable activeTab?: string; 
```
which would make sense, because health checks for this value are already present in some places, e.g. on line 42:
```
if (!this.activeTab && items.length > 0) {
```
Right now it seems a bit confusing to me.

Reply:
Oh I guess I should use `string | undefined` instead of `string | null`. Will do